### ISSUE TITLE
Shape of RaggedTensor is unknown when converting to tensor

### ISSUE THREAD

**Issue by tmattio** [tensorflow/tensorflow#31486]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS 10.14.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): From Pypa
- TensorFlow version (use command below): v1.14.0-rc1-22-gaf24dc91b5 1.14.0
- Python version: 3.6.4

**Describe the current behavior**

I am using the estimator API. I have variable-lenght input (sentences), so I want to use RaggedTensors in my `tf.data.Datasets`. However, it seems that when converting the RaggedTensor back to tensor, the shape is not evaluated correctly.

**Describe the expected behavior**

When using `x.to_tensor(...)` on the elements of the dataset, the shape should be evaluated to the shape of the tensor.

**Code to reproduce the issue**

```
import tensorflow as tf

shapes = [None]
types = tf.string
defaults = "<pad>"


def generator_fn_ragged():
    yield (
        ["The", "brown", "fox", "jumps"],
        ["The", "brown", "fox", "jumps", "over", "the", "lazy", "dog"],
    )


def input_fn():
    dataset = tf.data.Dataset.from_generator(
        generator_fn_ragged,
        output_shapes=shapes,
        output_types=tf.string,
    )
    dataset = dataset.padded_batch(2, shapes, defaults)
    
    def _ragged(*features):
        return [tf.RaggedTensor.from_tensor(x, padding="<pad>") for x in features]

    dataset = dataset.map(_ragged)

    def _unragged(*features):
        return [x.to_tensor(default_value="<pad>") for x in features]

    dataset = dataset.map(_unragged)
    
    return dataset

dataset = input_fn()
for el in dataset:
    print(el)
    break
```

This outputs:
```
(<tf.Tensor 'IteratorGetNext_7608:0' shape=<unknown> dtype=string>,)
```

Whereas this:
```
import tensorflow as tf

shapes = [None]
types = tf.string
defaults = "<pad>"


def generator_fn_ragged():
    yield (
        ["The", "brown", "fox", "jumps"],
        ["The", "brown", "fox", "jumps", "over", "the", "lazy", "dog"],
    )


def input_fn():
    dataset = tf.data.Dataset.from_generator(
        generator_fn_ragged,
        output_shapes=shapes,
        output_types=types,
    )
    dataset = dataset.padded_batch(2, shapes, defaults)
    return dataset

dataset = input_fn()
for el in dataset:
    print(el)
    break
```

Outputs:
```
Tensor("IteratorGetNext_4:0", shape=(?, ?), dtype=string)
```

**Comment by jvishnuvardhan** [comment]
@tmattio Can you provide a standalone code to reproduce the issue? Current code throws an error. Thanks!

**Comment by tmattio** [comment]
@jvishnuvardhan I just copy-pasted both of the code sample in a file and ran them without any issue.

Are you using the same environment? What is you error?

**Comment by jvishnuvardhan** [comment]
@tmattio Sorry for the mistake. I tried with `tf-nightly` and reported the error. Yes, I can reproduce the issue with `TF1,14`. Thanks!

**Comment by edloper** [comment]
It looks to me like this has already been fixed.  In particular, if I try your code sample in tf-nightly, then it outputs:

```
(<tf.Tensor 'IteratorGetNext_2:0' shape=(?, ?) dtype=string>,)
```

Though to make it work I needed to replace:

```
for el in dataset:
    print(el)
    break
```

With:

```
with tf.Session():
  it = dataset.make_one_shot_iterator()
  print it.get_next()
```

**Comment by edloper** [comment]
By the way, the PR that fixes this problem is 53fd64291fef36462293d634c84f2e338da6359d.
